check/all

```diff
@@ -31,8 +31,9 @@
 ################################################################################
 
 # Get the working directory to the repo root.
-cd "$(dirname "${BASH_SOURCE[0]}")"
```

Comment:
Why not `cd "$(dirname "${BASH_SOURCE[0]}")" || exit`?

My local shellcheck says `^-- SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.`

Reply:
That's what we are doing, i'm just curious about the creation of the `thisdir` variable.

Reply:
> Why not cd "$(dirname "${BASH_SOURCE[0]}")"

Such `cd` would still pass for a failing `$(...)` command.  If that command had no standard output,
the shell would expand it as `cd ""` and stay in the same working directory.
That could have unexpected effects further below.

Here is how to test -
```sh
$ cd /tmp
$ cd "$(false)" || echo "cd failed"
```

Reply:
> That's what we are doing, i'm just curious about the creation of the thisdir variable.

This allows us to fail when `$(...)` fails -
```
$ thisdir=$(false) || echo directory expression failed
```